src/main.rs

```diff
@@ -663,77 +798,269 @@ fn get_setup(args: &[String]) -> Setup {
-                                panic!("Invalid proxy url, only URLs on the format \"http://host:port\" are allowed");
+                                error!("Invalid proxy url, only URLs on the format \"http://host:port\" are allowed");
+                                exit(1);
                             }
 
                             if url.scheme() != "http" {
-                                panic!("Only unsecure http:// proxies are supported");
+                                error!("Only unsecure http:// proxies are supported");
+                                exit(1);
                             }
+
                             url
                         },
-                        Err(err) => panic!("Invalid proxy URL: {}, only URLs in the format \"http://host:port\" are allowed", err)
+                        Err(err) => {
+                            error!("Invalid proxy URL: {}, only URLs in the format \"http://host:port\" are allowed", err);
+                            exit(1);
+                        }
                     }
                 },
             ),
             ap_port: matches
                 .opt_str(AP_PORT)
-                .map(|port| port.parse::<u16>().expect("Invalid port")),
+                .map(|port| {
+                        let port = port.parse::<u16>().unwrap_or_else(|_| {
+                        error!("Invalid --{}: {}", AP_PORT, port);
+                        println!("Valid --{} values: 1-65535", AP_PORT);
+                        exit(1);
+                    });
+
+                    if port == 0 {
+                        error!("Invalid --{}: {}", AP_PORT, port);
+                        println!("Valid --{} values: 1-65535", AP_PORT);
+                        exit(1);
+                    }
+
+                    port
+                }),
         }
     };
 
     let player_config = {
         let bitrate = matches
             .opt_str(BITRATE)
             .as_deref()
-            .map(|bitrate| Bitrate::from_str(bitrate).expect("Invalid bitrate"))
-            .unwrap_or_default();
+            .map(|bitrate| {
+                Bitrate::from_str(bitrate).unwrap_or_else(|_| {
+                    error!("Invalid --{}: {}", BITRATE, bitrate);
+                    println!("Valid --{} values: 96, 160, 320", BITRATE);
+                    println!("Default: 160");
+                    exit(1);
+                })
+            })
+            .unwrap_or(PlayerConfig::default().bitrate);
 
         let gapless = !matches.opt_present(DISABLE_GAPLESS);
 
         let normalisation = matches.opt_present(ENABLE_VOLUME_NORMALISATION);
+
         let normalisation_method = matches
             .opt_str(NORMALISATION_METHOD)
             .as_deref()
             .map(|method| {
-                NormalisationMethod::from_str(method).expect("Invalid normalisation method")
+                warn!(
+                    "--{} will be deprecated in a future release.",
```

Comment:
While you're here, could you add a deprecation notice for the basic normaliser? Can be here or in `player.rs`.

Reply:
Will do. I like them here if that's OK?

Reply:
Done.